sdk/core/core-lro/src/poller/syncPoller.ts

```diff
@@ -0,0 +1,382 @@
+  setError: (state, error) => (state.error = error),
+  setResult: (state, result) => (state.result = result),
+  setRunning: (state) => (state.status = "running"),
+  setSucceeded: (state) => (state.status = "succeeded"),
+  setFailed: (state) => (state.status = "failed"),
+
+  getError: (state) => state.error,
+  getResult: (state) => state.result,
+  isCanceled: (state) => state.status === "canceled",
+  isFailed: (state) => state.status === "failed",
+  isRunning: (state) => state.status === "running",
+  isSucceeded: (state) => state.status === "succeeded",
+});
+
+/**
+ * ============================ Option 1: devired promise from pollUntilDone ============================
+ * step 0: init state with notStarted status and config it when calling initOperation
+ * Step 1: create poller
+ * Step 2: get result promise by calling poller.pollUntilDone()
+ * Step 3: somehow delegate the polling operations to poller and delegate promise operations to promise
+ * Step 4: return the poller & promise
+ * =====================================================================================================
+ */
+export function buildCreatePollerOption1<TResponse, TResult, TState extends OperationState<TResult>>(
+  inputs: BuildCreatePollerOptions<TResponse, TState>
+): (
+  lro: Operation<TResponse, { abortSignal?: AbortSignalLike }>,
+  options?: CreatePollerOptions<TResponse, TResult, TState>
+) => PromiseWithSimplePollerLike<TState, TResult> {
+  const {
+    getOperationLocation,
+    getStatusFromInitialResponse,
+    getStatusFromPollResponse,
+    isOperationError,
+    getResourceLocation,
+    getPollingInterval,
+    getError,
+    resolveOnUnsuccessful,
+  } = inputs;
+  return (
+    { init, poll }: Operation<TResponse, { abortSignal?: AbortSignalLike }>,
+    options?: CreatePollerOptions<TResponse, TResult, TState>
+  ) => {
+    const {
+      processResult,
+      updateState,
+      withOperationLocation: withOperationLocationCallback,
+      intervalInMs = POLL_INTERVAL_IN_MS,
+      restoreFrom,
+    } = options || {};
+    const stateProxy = createStateProxy<TResult, TState>();
+    const withOperationLocation = withOperationLocationCallback
+      ? (() => {
+        let called = false;
+        return (operationLocation: string, isUpdated: boolean) => {
+          if (isUpdated) withOperationLocationCallback(operationLocation);
+          else if (!called) withOperationLocationCallback(operationLocation);
+          called = true;
+        };
+      })()
+      : undefined;
+    const statePromise: Promise<RestorableOperationState<TState>> = restoreFrom
+      ? deserializeState(restoreFrom)
+      : initOperation({
+        init,
+        stateProxy,
+        processResult,
+        getOperationStatus: getStatusFromInitialResponse,
+        withOperationLocation,
+        setErrorAsResult: !resolveOnUnsuccessful,
+      });
+    let state: TState = {
+      status: "notStarted",
```

Comment:
Init the state with notStarted. Then reset the value when the promise one is resolved.